### PR TITLE
DON-707 Handle invalid slug

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

File: src/app/campaign.resolver.ts [hunk 1]
```diff
@@ -35,6 +38,16 @@ export class CampaignResolver implements Resolve<any> {
       );
     }
 
+    if (campaignSlug && fundSlug && campaignSlug !== "campaign") {
+      const query = this.campaignService.buildQuery(this.searchService.selected, 0, campaignId ?? undefined, campaignSlug, fundSlug);
+      const searchResult = this.campaignService.search(query as SearchQuery).subscribe(
+        () => {},
+        () => {
+          this.router.navigateByUrl(`/${campaignSlug}/`);
```
Comment (bdsl) [inline src/app/campaign.resolver.ts:46]: I'm not sure if this error handler will or is guaranteed to run during SSR

File: src/app/campaign.resolver.ts [hunk 2]
```diff
@@ -35,6 +38,16 @@ export class CampaignResolver implements Resolve<any> {
       );
     }
 
+    if (campaignSlug && fundSlug && campaignSlug !== "campaign") {
```
Comment (alihejazi) [inline src/app/campaign.resolver.ts:41]: Why do we need to check the following?
`campaignSlug !== "campaign"`

**Comment by alihejazi** [comment]
@bdsl I've pushed this commit: https://github.com/thebiggive/donate-frontend/pull/867/commits/5272d755cc1066103e2f24f7fd326b994f0339f6

File: src/app/campaign.resolver.ts (hunk 2 above)
Comment (bdsl) [reply src/app/campaign.resolver.ts:41]: My reading of the requriement was we only want to do this redirect when there's a specific metacampaign - e.g.

from https://donate.thebiggive.org.uk/christmas-challenge-2022/xxxxx to https://donate.thebiggive.org.uk/christmas-challenge-2022/ but not from https://donate.thebiggive.org.uk/campaign/abcd to https://donate.thebiggive.org.uk/campaign/

**Review by alihejazi — APPROVED** [review]

**Pull request merged**